pkg/proxy/proxy/transparent.go

```diff
@@ -67,21 +68,48 @@ func NewTransparentProxyServer(verbose bool) *goproxy.ProxyHttpServer {
 	return t
 }
 
+type ProxyMode int
+
+const (
+	MonitorMode ProxyMode = iota
+	EnforcementMode
+	UnkownMode
+)
+
+func StringToProxyMode(mode string) ProxyMode {
+	switch mode {
+	case "monitor":
+		return MonitorMode
+	case "enforce":
+		return EnforcementMode
+	default:
+		return UnkownMode
+	}
+}
+
 // TransparentProxyService transparently proxies HTTP and HTTPS traffic.
 type TransparentProxyService struct {
 	Proxy      *goproxy.ProxyHttpServer
 	Ca         *tls.Certificate
 	NetworkLog *netlog.NetworkActivityLog
-	mx         *sync.Mutex
+	Policy     *NetworkPolicy
+	Mode       ProxyMode
+
+	mx *sync.Mutex
 }
 
 // NewTransparentProxyService creates a new TransparentProxyService.
-func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate) TransparentProxyService {
+func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode string) TransparentProxyService {
 	m := new(sync.Mutex)
+	pm := StringToProxyMode(mode)
```

Comment:
okay so I think here it could be cool to do something closer to this: https://www.davidagood.com/go-enum-pattern/